src/snowflake/sqlalchemy/compat.py

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+from __future__ import annotations
+
+import functools
+import warnings
+from typing import Callable
+
+from sqlalchemy import __version__ as SA_VERSION
+from sqlalchemy import util
+
+string_types = (str,)
+returns_unicode = util.symbol("RETURNS_UNICODE")
+
+IS_VERSION_20 = tuple(int(v) for v in SA_VERSION.split(".")) >= (2, 0, 0)
+
+
+def args_reducer(positions_to_drop: tuple):
```

Comment:
1. Do I correctly understand it is sth like "args reducer for version < 2.0)? If yes, then is it possible to somehow include this info in the decorator name without making it too descriptive? The same question to `args_filler`.
2. Who is calling functions decorated with this decorator? I'm not sure if I understand why this decorator is needed.
3. Is this solution will work with keyword arguments? Does it have to?

Reply:
Signature of methods for ormselectcompilestate has changed - 2 parameters on position 6 and 7 were removed. This resulted in arguments loss. It does not need to work with anything else than tuple - as annotated.

This will be removed when we drop support for 1.4

The methods are called internally by orm engine when performing joins.

I'm open to suggestion for more convenient name 👍

Reply:
Let's stay with current name.